### DISCUSSION TITLE
Getting the following errors when trying to run the "pip install -r requirements.txt" command in MSYS MinGW64, how do I fix it?

### DISCUSSION THREAD

**Discussion by Route66Fan** [markjfine/nrsc5-dui#43] [Q&A]
R66F@DCDDCD-1QBKB3O5 MINGW64 /c/nrsc5-dui-master
$ pip install -r requirements.txt
Collecting pygobject==3.44.1 (from -r requirements.txt (line 1))
  Using cached PyGObject-3.44.1.tar.gz (720 kB)
  Installing build dependencies ... error
  error: subprocess-exited-with-error

  × pip subprocess to install build dependencies did not run successfully.
  │ exit code: 1
  ╰─> [67 lines of output]
      Collecting setuptools
        Using cached setuptools-75.3.0-py3-none-any.whl.metadata (6.9 kB)
      Collecting wheel
        Using cached wheel-0.44.0-py3-none-any.whl.metadata (2.3 kB)
      Collecting pycairo
        Using cached pycairo-1.27.0.tar.gz (661 kB)
        Installing build dependencies: started
        Installing build dependencies: finished with status 'done'
        Getting requirements to build wheel: started
        Getting requirements to build wheel: finished with status 'done'
        Installing backend dependencies: started
        Installing backend dependencies: finished with status 'error'
        error: subprocess-exited-with-error

        pip subprocess to install backend dependencies did not run successfully.
        exit code: 1

        [38 lines of output]
        Collecting ninja>=1.8.2
          Using cached ninja-1.11.1.1.tar.gz (132 kB)
          Installing build dependencies: started
          Installing build dependencies: finished with status 'done'
          Getting requirements to build wheel: started
          Getting requirements to build wheel: finished with status 'done'
          Preparing metadata (pyproject.toml): started
          Preparing metadata (pyproject.toml): finished with status 'done'
        Building wheels for collected packages: ninja
          Building wheel for ninja (pyproject.toml): started
          Building wheel for ninja (pyproject.toml): finished with status 'error'
          error: subprocess-exited-with-error

          Building wheel for ninja (pyproject.toml) did not run successfully.
          exit code: 1

          [15 lines of output]
          C:/msys64/tmp/pip-build-env-itfjaefp/overlay/lib/python3.11/site-packages/setuptools_scm/g
it.py:312: UserWarning: git archive did not support describe output
            warnings.warn("git archive did not support describe output")
          C:/msys64/tmp/pip-build-env-itfjaefp/overlay/lib/python3.11/site-packages/setuptools_scm/g
it.py:331: UserWarning: unprocessed git archival found (no export subst applied)
            warnings.warn("unprocessed git archival found (no export subst applied)")
          Traceback (most recent call last):
            File "C:/msys64/tmp/pip-build-env-itfjaefp/overlay/lib/python3.11/site-packages/skbuild/
setuptools_wrap.py", line 639, in setup
              cmkr = cmaker.CMaker(cmake_executable)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
            File "C:/msys64/tmp/pip-build-env-itfjaefp/overlay/lib/python3.11/site-packages/skbuild/
cmaker.py", line 145, in __init__
              self.cmake_version = get_cmake_version(self.cmake_executable)
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
            File "C:/msys64/tmp/pip-build-env-itfjaefp/overlay/lib/python3.11/site-packages/skbuild/
cmaker.py", line 102, in get_cmake_version
              raise SKBuildError(msg) from err

          Problem with the CMake installation, aborting build. CMake executable is cmake
          [end of output]

          note: This error originates from a subprocess, and is likely not a problem with pip.
          ERROR: Failed building wheel for ninja
        Failed to build ninja
        ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (nin
ja)
        [end of output]

        note: This error originates from a subprocess, and is likely not a problem with pip.
      error: subprocess-exited-with-error

      pip subprocess to install backend dependencies did not run successfully.
      exit code: 1

      See above for output.

      note: This error originates from a subprocess, and is likely not a problem with pip.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× pip subprocess to install build dependencies did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

**Comment by markjfine** [comment]
It's possible that my requirements.txt is out of date.
For example, I believe pyGObject is now up to 3.50.0 (at least per Homebrew). Try using that instead.

**Reply by Route66Fan** [reply]
Ok, I'll try that out. 

Thank you!

**Comment by markjfine** [comment]
I've just updated requirements.txt in Code.

**Comment by Route66Fan** [comment]
I tried using the updated requirements.txt file & now I am getting a different error message: 

R66F@DCDDCD-1QBKB3O5 MINGW64 /c/nrsc5-dui-master
$ pip install -r requirements.txt
Collecting pygobject==3.50.0 (from -r requirements.txt (line 1))
  Using cached pygobject-3.50.0.tar.gz (1.1 MB)
  Installing build dependencies ... error
  error: subprocess-exited-with-error

  × pip subprocess to install build dependencies did not run successfully.
  │ exit code: 1
  ╰─> [65 lines of output]
      Collecting meson-python>=0.12.1
        Using cached meson_python-0.17.1-py3-none-any.whl.metadata (4.1 kB)
      Collecting pycairo>=1.16
        Using cached pycairo-1.27.0.tar.gz (661 kB)
        Installing build dependencies: started
        Installing build dependencies: finished with status 'done'
        Getting requirements to build wheel: started
        Getting requirements to build wheel: finished with status 'done'
        Installing backend dependencies: started
        Installing backend dependencies: finished with status 'error'
        error: subprocess-exited-with-error

        pip subprocess to install backend dependencies did not run successfully.
        exit code: 1

        [38 lines of output]
        Collecting ninja>=1.8.2
          Using cached ninja-1.11.1.1.tar.gz (132 kB)
          Installing build dependencies: started
          Installing build dependencies: finished with status 'done'
          Getting requirements to build wheel: started
          Getting requirements to build wheel: finished with status 'done'
          Preparing metadata (pyproject.toml): started
          Preparing metadata (pyproject.toml): finished with status 'done'
        Building wheels for collected packages: ninja
          Building wheel for ninja (pyproject.toml): started
          Building wheel for ninja (pyproject.toml): finished with status 'error'
          error: subprocess-exited-with-error

          Building wheel for ninja (pyproject.toml) did not run successfully.
          exit code: 1

          [15 lines of output]
          C:/msys64/tmp/pip-build-env-1idv60rk/overlay/lib/python3.11/site-packages/setuptools_scm/g
it.py:312: UserWarning: git archive did not support describe output
            warnings.warn("git archive did not support describe output")
          C:/msys64/tmp/pip-build-env-1idv60rk/overlay/lib/python3.11/site-packages/setuptools_scm/g
it.py:331: UserWarning: unprocessed git archival found (no export subst applied)
            warnings.warn("unprocessed git archival found (no export subst applied)")
          Traceback (most recent call last):
            File "C:/msys64/tmp/pip-build-env-1idv60rk/overlay/lib/python3.11/site-packages/skbuild/
setuptools_wrap.py", line 639, in setup
              cmkr = cmaker.CMaker(cmake_executable)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
            File "C:/msys64/tmp/pip-build-env-1idv60rk/overlay/lib/python3.11/site-packages/skbuild/
cmaker.py", line 145, in __init__
              self.cmake_version = get_cmake_version(self.cmake_executable)
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
            File "C:/msys64/tmp/pip-build-env-1idv60rk/overlay/lib/python3.11/site-packages/skbuild/
cmaker.py", line 102, in get_cmake_version
              raise SKBuildError(msg) from err

          Problem with the CMake installation, aborting build. CMake executable is cmake
          [end of output]

          note: This error originates from a subprocess, and is likely not a problem with pip.
          ERROR: Failed building wheel for ninja
        Failed to build ninja
        ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (nin
ja)
        [end of output]

        note: This error originates from a subprocess, and is likely not a problem with pip.
      error: subprocess-exited-with-error

      pip subprocess to install backend dependencies did not run successfully.
      exit code: 1

      See above for output.

      note: This error originates from a subprocess, and is likely not a problem with pip.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× pip subprocess to install build dependencies did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

**Comment by markjfine** [comment]
Actually, this could be several errors in pip as well as python - and I'm learning to love python as much as I 'love' java from an ease of maintenance standpoint. Python's already up to 3.13, as I just found out by upgrading to Fedora 41.

As such, I can almost guarantee that things are deprecated out of Python 3.11, as much as they may not be ready for primetime in 3.13. This was a problem when 3.12 was introduced (had no python-dateutils available), and may be a problem again. Try building gr-osmosdr when Linux has moved you to 3.13, and the pybind11 makefile rigidly needs the includes for 3.12.

More specifically in your case, It looks like ninja has an issue building in 3.11. You could try killing 3.11 and moving to 3.12, then reinstalling all the dependencies there, or setting up a virtual python environment via venv.

Secondly, I see you're trying to do this under MSYS in Windows, which I've never gotten to work reliably.
